verification/src/http_server/mod.rs

```diff
@@ -1,22 +1,50 @@
 pub mod handlers;
+pub mod metrics;
 mod routers;
 
 pub use self::routers::{configure_router, AppRouter, Router};
 
 use crate::config::Config;
 use actix_web::{App, HttpServer};
+
+use futures::future;
+use metrics::Metrics;
 use std::sync::Arc;
 
 pub async fn run(config: Config) -> std::io::Result<()> {
     let socket_addr = config.server.addr;
+    let metrics_enabled = config.metrics.enabled;
+    let metrics_addr = config.metrics.addr;
+    let metrics_endpoint = config.metrics.endpoint.clone();
+
     log::info!("Verification server is starting at {}", socket_addr);
     let app_router = Arc::new(
         AppRouter::new(config)
             .await
             .expect("couldn't initialize the app"),
     );
-    HttpServer::new(move || App::new().configure(configure_router(&*app_router)))
+    let metrics = Metrics::new(metrics_endpoint);
```

Comment:
can we just not create this if `metrics_enabled == false` ?

Reply:
we need this value in `server_future `